vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/Main.java

```diff
@@ -745,7 +746,35 @@ protected Optional<Target> targetByName(String name) {
           .select(WORKFLOW.NAME, WORKFLOW.MAX_IN_FLIGHT)
           .from(WORKFLOW)
           .forEach(record -> maxInFlightPerWorkflow.set(record.value1(), record.value2()));
+
+      DSL.using(connection)
+          .select(WORKFLOW.NAME, WORKFLOW_RUN.HASH_ID,
+              DSL.jsonObject(ACTIVE_WORKFLOW_RUN.CONSUMABLE_RESOURCES))
+          .from(WORKFLOW)
+          .join(WORKFLOW_VERSION)
+          .on(WORKFLOW.NAME.eq(WORKFLOW_VERSION.NAME))
+          .join(WORKFLOW_RUN)
+          .on(WORKFLOW_RUN.WORKFLOW_VERSION_ID.eq(WORKFLOW_VERSION.ID))
+          .join(ACTIVE_WORKFLOW_RUN)
+          .on(WORKFLOW_RUN.ID.eq(ACTIVE_WORKFLOW_RUN.ID))
+          .where(ACTIVE_WORKFLOW_RUN.ENGINE_PHASE.eq(Phase.WAITING_FOR_RESOURCES))
+          .forEach(record -> {
+            try {
+              priorityPerWorkflow.set(record.value1(),
+                  record.value2(),
+                  MAPPER.readTree((record.value3() == null || record.value3().data() == null) ?
+                      "{}" :
+                      record.value3().data()));
+            } catch (JsonProcessingException e) {
+              // not a disaster; we might just get some things running out of priority
+              // until max-in-flight gets saturated
+              System.out.println("Uh oh, failed to serialize the consumable resources "
+                  + "field on active workflow run for priority by workflow on startup:");
```

Comment:
What do you think about adding a line here like `Some workflow runs might run out of priority order until the workflow's max-in-flight is saturated` to indicate what effects this will have?